### ISSUE TITLE
conda create ..... --dry-run option fails when --clone is used

### ISSUE THREAD

**Issue by marshall245** [conda/conda#2470]
### Notes

Using the --dry-run option fails when used in conjunction with the --clone option. The new environment is still created. Using it without the --clone option appears to still work.
##### With --clone

``` bash
mmarkham@VM112:~$ conda env list
# conda environments:
#
_test                    /home/mmarkham/anaconda2/envs/_test
working                  /home/mmarkham/anaconda2/envs/working
root                  *  /home/mmarkham/anaconda2

mmarkham@VM112:~$ clear
mmarkham@VM112:~$ conda env list
# conda environments:
#
_test                    /home/mmarkham/anaconda2/envs/_test
working                  /home/mmarkham/anaconda2/envs/working
root                  *  /home/mmarkham/anaconda2

mmarkham@VM112:~$ conda create -n an_env --clone working --dry-run
Fetching package metadata: ............
src_prefix: '/home/mmarkham/anaconda2/envs/working'
dst_prefix: '/home/mmarkham/anaconda2/envs/an_env'
Packages: 188
Files: 37
Linking packages ...
[      COMPLETE      ]|###################################################| 100%
#
# To activate this environment, use:
# $ source activate an_env
#
# To deactivate this environment, use:
# $ source deactivate
#
mmarkham@VM112:~$ conda env list
# conda environments:
#
_test                    /home/mmarkham/anaconda2/envs/_test
an_env                   /home/mmarkham/anaconda2/envs/an_env
working                  /home/mmarkham/anaconda2/envs/working
root                  *  /home/mmarkham/anaconda2
```
##### Without --clone

``` bash
mmarkham@VM112:~$ conda create -n an_env python=2 --dry-run
Fetching package metadata: ............
Solving package specifications: .........

Package plan for installation in environment /home/mmarkham/anaconda2/envs/an_env:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2h-0
    pip:        8.1.1-py27_1
    python:     2.7.11-0
    readline:   6.2-2
    setuptools: 20.7.0-py27_0
    sqlite:     3.9.2-0
    tk:         8.5.18-0
    wheel:      0.29.0-py27_0
    zlib:       1.2.8-0

Dry run: exiting
mmarkham@VM112:~$ conda env list
# conda environments:
#
_test                    /home/mmarkham/anaconda2/envs/_test
working                  /home/mmarkham/anaconda2/envs/working
root                  *  /home/mmarkham/anaconda2
```

**Comment by ijstokes** [comment]
Possibly linked to #1597 

@mcg1969 another `--clone` issue, although unclear what Conda version is in use.

@marshall245 could you give us the output of `conda info -a` (in fact `-a` may be overkill).

**Comment by marshall245** [comment]
```bash
mmarkham@VM112:~$ conda info -a
Current conda install:

             platform : linux-64
        conda version : 4.0.6
  conda-build version : 1.18.1
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /home/mmarkham/anaconda2  (writable)
  default environment : /home/mmarkham/anaconda2
     envs directories : /home/mmarkham/anaconda2/envs
        package cache : /home/mmarkham/anaconda2/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
                        https://conda.anaconda.org/melund/linux-64/
                        https://conda.anaconda.org/melund/noarch/
                        http://qr-01:8888/nest/linux-64/
                        http://qr-01:8888/nest/noarch/
                        http://conda.maxpoint.net/incubator/linux-64/
                        http://conda.maxpoint.net/incubator/noarch/
                        http://10.10.3.112:1221/pkgs/linux-64/
                        http://10.10.3.112:1221/pkgs/noarch/
          config file : /home/mmarkham/.condarc
    is foreign system : False

# conda environments:
#
_test                    /home/mmarkham/anaconda2/envs/_test
working                  /home/mmarkham/anaconda2/envs/working
root                  *  /home/mmarkham/anaconda2

sys.version: 2.7.11 |Anaconda 2.4.0 (64-bit)| (defaul...
sys.prefix: /home/mmarkham/anaconda2
sys.executable: /home/mmarkham/anaconda2/bin/python
conda location: /home/mmarkham/anaconda2/lib/python2.7/site-packages/conda
conda-build: /home/mmarkham/anaconda2/bin/conda-build
conda-convert: /home/mmarkham/anaconda2/bin/conda-convert
conda-develop: /home/mmarkham/anaconda2/bin/conda-develop
conda-env: /home/mmarkham/anaconda2/bin/conda-env
conda-index: /home/mmarkham/anaconda2/bin/conda-index
conda-inspect: /home/mmarkham/anaconda2/bin/conda-inspect
conda-metapackage: /home/mmarkham/anaconda2/bin/conda-metapackage
conda-pipbuild: /home/mmarkham/anaconda2/bin/conda-pipbuild
conda-server: /home/mmarkham/anaconda2/bin/conda-server
conda-sign: /home/mmarkham/anaconda2/bin/conda-sign
conda-skeleton: /home/mmarkham/anaconda2/bin/conda-skeleton
user site dirs:

CIO_TEST: <not set>
CONDA_DEFAULT_ENV: <not set>
CONDA_ENVS_PATH: <not set>
LD_LIBRARY_PATH: <not set>
PATH: /home/mmarkham/anaconda2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
PYTHONHOME: <not set>
PYTHONPATH: <not set>

WARNING: could not import _license.show_info
# try:
# $ conda install -n root _license
```

**Comment by TCAllen07** [comment]
I ran into the same issue just now, trying to test the transition of a project from Python 2 to 3 on macOS. To confirm, I reproduced the steps on an AWS instance with the same result. I thought I'd attach some additional details/output for comparison.

### macOS dry-run clone & conda info
```
trevor@nikola:mtxrules$ conda create --dry-run --name mtx2bak --clone mtxenv
Source:      /Users/trevor/anaconda2/envs/mtxenv
Destination: /Users/trevor/anaconda2/envs/mtx2bak
Using Anaconda API: https://api.anaconda.org
Packages: 64
Files: 127
Linking packages ...
[      COMPLETE      ]|###############################################################################################################################################################################| 100%
#
# To activate this environment, use:
# $ source activate mtx2bak
#
# To deactivate this environment, use:
# $ source deactivate
#
trevor@nikola:mtxrules$ conda create --name mtx2bak --clone mtxenv
Error: prefix already exists: /Users/trevor/anaconda2/envs/mtx2bak
trevor@nikola:mtxrules$ conda env list
# conda environments:
#
matchtxdemo              /Users/trevor/anaconda2/envs/matchtxdemo
mtx2bak                  /Users/trevor/anaconda2/envs/mtx2bak
mtx3                     /Users/trevor/anaconda2/envs/mtx3
mtxenv                   /Users/trevor/anaconda2/envs/mtxenv
npd                      /Users/trevor/anaconda2/envs/npd
root                  *  /Users/trevor/anaconda2

trevor@nikola:mtxrules$ conda info
Current conda install:

             platform : osx-64
        conda version : 4.1.12
    conda-env version : 2.5.2
  conda-build version : 2.0.7
       python version : 2.7.12.final.0
     requests version : 2.10.0
     root environment : /Users/trevor/anaconda2  (writable)
  default environment : /Users/trevor/anaconda2
     envs directories : /Users/trevor/anaconda2/envs
        package cache : /Users/trevor/anaconda2/pkgs
         channel URLs : https://conda.anaconda.org/t/<TOKEN>/conda-forge/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/conda-forge/noarch/
                        https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /Users/trevor/.condarc
         offline mode : False
    is foreign system : False
```

### Ubuntu dry-run clone & conda info
```
ubuntu@ip-172-31-61-80:~$ conda env list
# conda environments:
#
mtxenv                   /home/ubuntu/miniconda2/envs/mtxenv
root                  *  /home/ubuntu/miniconda2

ubuntu@ip-172-31-61-80:~$ conda update conda
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /home/ubuntu/miniconda2:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.2.13               |           py27_0         391 KB

The following packages will be UPDATED:

    conda: 4.2.12-py27_0 --> 4.2.13-py27_0

Proceed ([y]/n)? y

Fetching packages ...
conda-4.2.13-p 100% |#######################################################################################| Time: 0:00:00  19.60 MB/s
Extracting packages ...
[      COMPLETE      ]|##########################################################################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|##########################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|##########################################################################################################| 100%
ubuntu@ip-172-31-61-80:~$ conda create --dry-run --name testmtx2bak --clone mtxenv
Source:      /home/ubuntu/miniconda2/envs/mtxenv
Destination: /home/ubuntu/miniconda2/envs/testmtx2bak
Packages: 73
Files: 126
Fetching package metadata .....
Pruning fetched packages from the cache ...
Fetching packages ...
decorator-4.0. 100% |#######################################################################################| Time: 0:00:00  16.20 MB/s
ipython_genuti 100% |#######################################################################################| Time: 0:00:00  14.71 MB/s
ptyprocess-0.5 100% |#######################################################################################| Time: 0:00:00  12.36 MB/s
wcwidth-0.1.7- 100% |#######################################################################################| Time: 0:00:00  20.19 MB/s
pathlib2-2.1.0 100% |#######################################################################################| Time: 0:00:00   3.91 MB/s
ipython-5.1.0- 100% |#######################################################################################| Time: 0:00:00  16.31 MB/s
Extracting packages ...
[      COMPLETE      ]|##########################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|##########################################################################################################| 100%
#
# To activate this environment, use:
# > source activate testmtx2bak
#
# To deactivate this environment, use:
# > source deactivate testmtx2bak
#

ubuntu@ip-172-31-61-80:~$ conda env list
# conda environments:
#
mtxenv                   /home/ubuntu/miniconda2/envs/mtxenv
testmtx2bak              /home/ubuntu/miniconda2/envs/testmtx2bak
root                  *  /home/ubuntu/miniconda2

ubuntu@ip-172-31-61-80:~$ conda info
Current conda install:

               platform : linux-64
          conda version : 4.2.13
       conda is private : False
      conda-env version : 4.2.13
    conda-build version : not installed
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : /home/ubuntu/miniconda2  (writable)
    default environment : /home/ubuntu/miniconda2
       envs directories : /home/ubuntu/miniconda2/envs
          package cache : /home/ubuntu/miniconda2/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
           offline mode : False
```

**Comment by kalefranz** [comment]
This will no longer happen with #5201.  We could now fail with an ugly stack trace assertion error, but we can deal with that if we ever encounter it.  Ideally it's all already fixed and #5201 is just extra enforcement.
